Add documentation for how to integrate v2 with external bundlers and transpilers. (#461)
mapbox/mapbox-gl-js-docs: docs/pages/api/index.md

  * *Terrain:* The height map for over-zoomed tiles is not being retained.
  * To report new issues with Mapbox GL JS v2, create a [bug report](https://github.com/mapbox/mapbox-gl-js/issues/new?template=Bug_report.md) on GitHub.

## Transpiling v2

> Mapbox GL JS v2 is distributed as an ES6 compatible JavaScript bundle and is compatible with all major modern browsers. If you are using v2 with a module bundler such as Webpack or Rollup along with a transpiler such as Babel, then Babel needs to be configured to skip transpilation of Mapbox GL JS.
> Configure the [ignore](https://babeljs.io/docs/en/options#ignore) option in Babel to do this.

Reviewer: ```suggestion
Mapbox GL JS v2 is distributed as an ES6 compatible JavaScript bundle and is compatible with all major modern browsers. If you are using v2 with a module bundler such as Webpack or Rollup along with a transpiler such as Babel, use the ignore option in Babel to prevent Mapbox GL JS from being transpiled:
```

```
ignore: [ './node_modules/mapbox-gl/mapbox-gl.js' ]
```

> If your application requires ES5 compatibility, then your module bundler needs to be configured to load and transpile Mapbox GL JS's WebWorker separately. Mapbox GL JS can be configured with bundler specific `worker-loader` plugins. (See. [webpack-worker-loader] (https://webpack.js.org/loaders/worker-loader/) and [rollup-plugin-worker-loader](https://www.npmjs.com/package/rollup-plugin-web-worker-loader)). The following snippets shows integration with Webpack:

Reviewer: ```suggestion
If your application requires ES5 compatibility, configure your bundler to load and transpile Mapbox GL JS's WebWorker separately. You can configure Mapbox GL JS with bundler specific `worker-loader` plugins such as [webpack-worker-loader] (https://webpack.js.org/loaders/worker-loader/) and [rollup-plugin-worker-loader](https://www.npmjs.com/package/rollup-plugin-web-worker-loader). 
```

> * Setup `webpack.config.js` to use `worker-loader`:
> ```

Reviewer: ```suggestion
Setup `webpack.config.js` to use `worker-loader`:

```js
```

module.exports = {
  module: {
    rules: [
      {
        test: /\bmapbox-gl-csp-worker.js\b/i,
        use: { loader: "worker-loader" },
      },
    ],
  },
};
```

> * Integrate the Webpack loaded worker with mapbox-gl:
> ```

Reviewer: ```suggestion
Integrate the Webpack loaded worker with Mapbox GL JS:

```js
```

  * *Terrain:* The height map for over-zoomed tiles is not being retained.
  * To report new issues with Mapbox GL JS v2, create a [bug report](https://github.com/mapbox/mapbox-gl-js/issues/new?template=Bug_report.md) on GitHub.

> ## Transpiling v2

Reviewer 2: Are there other steps that developers need to follow when using GL-JS with Modules bundlers? Maybe include that and use a more generic title so that it's easy for getting started.

Author: No, the issue is only with bundlers + transpilation, hence the title.
